src/sparse_int_set.jl

```diff
@@ -226,24 +226,25 @@ function in_excluded(id, it)
     return false
 end
 
-@inline function id_tids(it, state)
-    id = it.shortest_set.packed[state]
-    return id, map(x -> findfirst_packed_id(id, x), it.valid_sets)
+@inline function in_valid(id, it)
+    for e in it.valid_sets
+        if !in(id, e)
+            return false
+        end
+    end
+    return true
 end
 
-Base.@propagate_inbounds function iterate(it::ZippedSparseIntSetIterator, state=1)
+@inline function iterate(it::ZippedSparseIntSetIterator, state=1)
     iterator_length = length(it)
     if state > iterator_length
         return nothing
     end
-    id, tids = id_tids(it, state)
-    while any(iszero, tids) || in_excluded(id, it)
-        state += 1
-        if state > iterator_length
-            return nothing
-        end
-
-        id, tids = id_tids(it, state)
-    end
-    return tids, state + 1
+    for i in state:iterator_length
+	    @inbounds id = it.shortest_set.packed[i]
+		if in_valid(id, it) && !in_excluded(id, it)
+			return map(x->findfirst_packed_id(id, x), it.valid_sets), i + 1
+		end
+	end
+	return nothing
```

Comment:
indenting seems to have flipped over to tabs again 😂

Reply:
oh yes, dammit! I need to figure out why that happens 😂